elliott/elliottlib/cli/attach_cve_flaws_cli.py

```diff
@@ -1,23 +1,390 @@
+
+        if not shipment_config:
+            raise ValueError(f"No shipment config found for kind: {self.default_advisory_type}")
+
+        release_notes = shipment_config.shipment.data.releaseNotes
+        if not release_notes:
+            raise ValueError(f"No release notes found in shipment config for {self.default_advisory_type}")
+
+        return release_notes
+
+    def get_attached_trackers(self, bugs_ids: List[str], bug_tracker: BugTracker) -> List[Bug]:
+        """
+        Get attached tracker bugs from a list of bug IDs.
+        """
+
+        if not bugs_ids:
+            return []
+
+        attached_tracker_bugs: List[Bug] = bug_tracker.get_tracker_bugs(bugs_ids)
+        if not attached_tracker_bugs:
+            self.logger.info(f'Found 0 {bug_tracker.type} tracker bugs attached')
+            return []
+
+        self.logger.info(
+            f'Found {len(attached_tracker_bugs)} {bug_tracker.type} tracker bugs attached: '
+            f'{sorted([b.id for b in attached_tracker_bugs])}'
+        )
+        return attached_tracker_bugs
+
+    def update_konflux_release_notes_to_rhsa(
+        self,
+        release_notes: ReleaseNotes,
+        flaw_bugs: Iterable[Bug],
+        tracker_bugs: List[Bug],
+        tracker_flaws: Dict[int, Iterable],
+    ):
+        """
+        Update the release notes to convert it to an RHSA type. Also adds CVE associations and flaw bugs.
+        """
+
+        if release_notes.type != 'RHSA':
+            # Set the release notes type to RHSA
+            self.logger.info("Converting release notes to RHSA type.")
+            release_notes.type = 'RHSA'
+
+        # Add the CVE component mapping to the cve field
+        cve_component_mapping = self.get_cve_component_mapping(flaw_bugs, tracker_bugs, tracker_flaws)
+        for cve_id, components in cve_component_mapping.items():
+            for component in components:
+                release_notes.cves.append(CveAssociation(key=cve_id, component=component))
+
+        # Attach the flaw bugs to the release notes issues
+        release_notes.issues.fixed.extend([{'id': bug.id, 'source': "bugzilla.redhat.com"} for bug in flaw_bugs])
+
+        # Update synopsis, topic and solution
+        cve_boilerplate = get_advisory_boilerplate(
+            runtime=self.runtime, et_data=self.errata_config, art_advisory_key=self.advisory_kind, errata_type='RHSA'
+        )
+        release_notes.synopsis = cve_boilerplate['synopsis'].format(MINOR=self.minor, PATCH=self.patch)
+        highest_impact = get_highest_security_impact(flaw_bugs)
+        release_notes.topic = cve_boilerplate['topic'].format(IMPACT=highest_impact, MINOR=self.minor, PATCH=self.patch)
+        release_notes.solution = cve_boilerplate['solution'].format(MINOR=self.minor)
+
+        # Add CVE component mapping
+        cve_component_mapping = self.get_cve_component_mapping(flaw_bugs, tracker_bugs, tracker_flaws)
```

Comment:
This block looks to be repeated from line 182?

Reply:
right, removed :+1: